src/toolkitAPI.c

```diff
@@ -1248,6 +1248,69 @@ int DLLEXPORT swmm_getSystemRunoffStats(SM_RunoffTotals *runoffTot)
     return(errorcode);
 }
 
+int DLLEXPORT swmm_getGagePrecip(int index, double *rainfall, double *snowfall, double *total)
+//
+// Input:   index = Index of desired ID
+// Output:  value = value to be output
+// Return:  API Error
+// Purpose: Gets the precipitaion value in the gage. 
+{
+    int errcode = 0;
+    // Check if Open
+    if(swmm_IsOpenFlag() == FALSE)
+    {
+	    errcode = ERR_API_INPUTNOTOPEN;
+    }
+    // Check if object index is within bounds
+    else if (index < 0 || index >= Nobjects[GAGE])
+    {
+	    errcode = ERR_API_OBJECT_INDEX;
+    }
+    // Read the rainfall value
+    else
+    {
+	    *rainfall = 0.0;
```

Comment:
@abhiramm7, Last change I promise!!!!

instead of duplicating functionality inside swmm, I suggest calling the following function directly

```C
double gage_getPrecip(int j, double *rainfall, double *snowfall)
//
//  Input:   j = rain gage index
//  Output:  rainfall = rainfall rate (ft/sec)
//           snowfall = snow fall rate (ft/sec)
//           returns total precipitation (ft/sec)
//  Purpose: determines whether gage's recorded rainfall is rain or snow.
//
{
    *rainfall = 0.0;
    *snowfall = 0.0;
    if ( !IgnoreSnowmelt && Temp.ta <= Snow.snotmp )
    {
       *snowfall = Gage[j].rainfall * Gage[j].snowFactor / UCF(RAINFALL);
    }
    else *rainfall = Gage[j].rainfall / UCF(RAINFALL);
    return (*rainfall) + (*snowfall);
} 
```

```C
*total = gage_getPrecip(int j, double *rainfall, double *snowfall);
```
and converting them:

```C
*rainfall /= UCF(RAINFALL);
*snowfall /= UCF(RAINFALL);
*total /= UCF(RAINFALL);
```

Reply:
Is gage_getPrecip not already converting the rain fall ?

Reply:
Maybe so!

Reply:
The rainfall is converted (for example) from mm to ft/sec internally in gage_getPrecip and then for purposes of continuity tables, tables and graphs converted back to mm.  I don't like the fact that the units of SWMM are USA units.  They should be SI units as eventually the USA will switch to SI units.

Reply:
@dickinsonre, didn’t they try this in the 70’s but ‘mercians were too stubborn? :-)